packages/core/src/lib/modal.svelte

```diff
@@ -73,15 +75,29 @@ $: if (typeof document !== 'undefined') {
   document.body.classList.toggle('overflow-hidden', isOpen);
 }
 
-$: headingElement?.focus();
+// Focus Logic
+// If one slot is passed focus slot element
+// if two slots are passed reference focusPrimaryElement and focus primary or secondary slot
+$: {
+  if (isOpen) {
+    const slotValues = slotElement?.querySelectorAll('button');
+    if (slotValues?.length !== undefined && slotValues.length > 0) {
+      if (focusPrimaryElement) {
+        slotValues.item(1).focus();
+      } else {
+        slotValues.item(0).focus();
+      }
+    }
+  }
+}
```

Comment:
I'm not sure about adding this logic to the Prime modal; the `querySelector` is a bit of a smell. It looks like our `<Button>` already accepts an `autofocus` prop via `$$restProps`

Can we trim this PR down to remove _all_ focus handling from the `<Modal>` to allow the end-user to make the decision per call-site?

Reply:
yeah I think I got a bit too invested in having prime solve all of a users focus cases.

Now that the modal focus logic is gone autofocus totally works in app. 

Removing all the focus stuff